packages/pg/lib/index.js

```diff
@@ -34,31 +34,37 @@ const PG = function (clientConstructor) {
   this.utils = utils
 }
 
-if (typeof process.env.NODE_PG_FORCE_NATIVE !== 'undefined') {
-  module.exports = new PG(require('./native'))
-} else {
-  module.exports = new PG(Client)
+let clientConstructor = Client
 
-  // lazy require native module...the native module may not have installed
-  Object.defineProperty(module.exports, 'native', {
-    configurable: true,
-    enumerable: false,
-    get() {
-      let native = null
-      try {
-        native = new PG(require('./native'))
-      } catch (err) {
-        if (err.code !== 'MODULE_NOT_FOUND') {
-          throw err
-        }
+try {
+  if (process.env.NODE_PG_FORCE_NATIVE) {
+    clientConstructor = require('./native')
+  }
+} catch {
+  // ignore, e.g., Deno without --allow-env
```

Comment:
Or `require('./native')` failed. The `try` should only cover the `process.env` access (if we actually want to do it this way at all).